iceoryx2/tests/node_tests.rs

```diff
@@ -318,6 +318,26 @@ mod node {
         assert_that!(node_counter, eq 1);
     }
 
+    #[test]
+    fn i_am_not_dead<S: Service>() {
+        let node = NodeBuilder::new().create::<S>().unwrap();
+
+        let mut nodes = vec![];
+        let result = Node::<S>::list(node.config(), |node_state| {
+            nodes.push(node_state);
+            CallbackProgression::Continue
+        });
+
+        assert_that!(result, is_ok);
+        assert_that!(nodes, len 1);
+
+        if let NodeState::Alive(node_view) = &nodes[0] {
+            assert_that!(node_view.id(), eq node.id());
+        } else {
+            assert_that!(unreachable);
```

Comment:
This is extremely confusing. In a few weeks, when reading this I will wonder why this works because the unreachable macro is used in a wrong way.
I would either use the unreachable/panic macro directly, use a different name for this usage or create a separate macro, e.g. assert_fail